src/nccl_ofi_net.c

```diff
@@ -1237,7 +1237,14 @@ static ncclResult_t ofi_getProperties(int dev, ncclNetProperties_t *props)
 		goto exit;
 	}
 
-	dev_props.name = strdup(nic_info->device_attr->name);
+	/* name is NULL if device is a part of multirail config */
+	if (nic_info->device_attr->name) {
+		dev_props.name = strdup(nic_info->device_attr->name);
+	} else {
+		dev_props.name = malloc(NAME_MAX);
+		snprintf(dev_props.name, NAME_MAX, "%s, device %d", nic_prov->domain_attr->name, dev);
```

Comment:
Given that we are fixing null names, should there be a check for a possible NULL in the provider name?

Reply:
Frankly to say it is also blindly strduped in set_nic_props_default() so if NULL there happened, one will see segfault from there. As nobody have seen it yet, I conclude that it is safe to use it without check here.

Although if you guys really want it, I can definitely add extra check. Shall I?

Reply:
Hello @nzmsv @rashikakheria, any updates on this? Shall I add one more check or is pull request fine as is?

Reply:
I just noticed that there is some code on the `aws` branch that assumes the name is at most `FI_NAME_MAX` long.  Could we use that for the allocation?

Reply:
Other than that - looks good.  I tested this for sanity also.

Reply:
Implemented approach with FI_NAME_MAX. Slightly restructured printf format string -- assuming that domain name itself can be FI_NAME_MAX long, it looks reasonable to put device number first, so if truncated, at least returned names are distinguishable.

Reply:
Thank you!  Could you squash this all into one commit and write up a message?  I was thinking something like:
```
Fix crash in case the underlying provider exposes multirail devices

Multirail devices commonly have a NULL name.  Do not assume that a name
has been provided by Libfabric and generate one if it is missing.
```

Reply:
Squashed, message changed. Removed generation as per request from @rashika (please see below).